zerver/data_import/slack.py

```diff
@@ -367,13 +398,11 @@ def get_user_email(user: ZerverFieldsT, domain_name: str) -> str:
     if user["is_mirror_dummy"]:
         return "{}@{}.slack.com".format(user["name"], user["team_domain"])
     if "bot_id" in user["profile"]:
-        if "real_name_normalized" in user["profile"]:
-            slack_bot_name = user["profile"]["real_name_normalized"]
-        elif "first_name" in user["profile"]:
-            slack_bot_name = user["profile"]["first_name"]
+        slack_bot_id = user["profile"]["bot_id"]
+        if slack_bot_id in SlackBotEmail.assigned_email:
+            return SlackBotEmail.assigned_email[slack_bot_id]
         else:
-            raise AssertionError("Could not identify bot type")
-        return slack_bot_name.replace("Bot", "").replace(" ", "") + f"-bot@{domain_name}"
+            return SlackBotEmail.generate_email(user["profile"], domain_name)
```

Comment:
I think it might be a cleaner interface to have this if/else live in the `SlackBotEmail` class, so you just call `return `SlackBotEmail.email_for(user)`